app/services/cleanup_service.py

```diff
@@ -0,0 +1,54 @@
+from sqlalchemy.orm import Session
+from app.models import UserModel, CourseModel
+from app.services import GiteaService, KubernetesService
+
+class CleanupService:
```

Comment:
Because cleanup has to be done in stages based on how far along the service method got in its execution before encountering an unrecoverable failure. The create assignment method only has 1 stage of cleanup (database deletion) and so does not need a whole method dedicated to cleanup.